### PR TITLE
:sparkles: factory / simple (array, string, bool)

### LINKED ISSUE #92: Rename simpleObject for simpleValue
The name `simpleObject` is misleading. We expect that the value backed by this factory have to be an object. Perhaps we can make it more obvious by naming it `simpleValue` which more clearly indicate that this factory is backing any kind of value.

### LINKED ISSUE #98: API / simpleObject / defaultValue
What do you think to add the possibility to add defaultValue with the first parameter ?

|function|defaultValue|
|--|--|
|simpleObject()|{}|
|simpleObject({ defaultValue: {} })|{}|
|simpleObject({ my: 'object' })|{}|
|simpleObject('string')|'string'|
|simpleObject([1, 2, 3])|[1, 2, 3]|

### PR DESCRIPTION
**Pull request by guillaumecrespel**
fixes #98
fixes #92

- [x] :memo:

### CODE REVIEW

File: src/__snapshots__/index.spec.js.snap [hunk 1]
```diff
@@ -12,6 +12,18 @@ exports[`index named factory should instanciate a keyValue reducer (specified) 1
...
 
+exports[`index suffix of simple factory should make a simple array init to empty 1`] = `Array []`;
+
+exports[`index suffix of simple factory should make a simple boolean init to false 1`] = `false`;
+
+exports[`index suffix of simple factory should make a simple object init to empty 1`] = `Object {}`;
+
+exports[`index suffix of simple factory should make a simple string init to blank 1`] = `""`;
+
+exports[`index suffix of simpleObject factory should make a simple object init to empty 1`] = `undefined`;
```
Comment (fabienjuif) [inline src/__snapshots__/index.spec.js.snap]: This one should not be possible into redux

File: src/index.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ const getWrappedStore = (middlewares = {}) => (options = {}) => {
     reducer({ ...middlewares, engine: typeConfig.middlewares })(key)(prefix)(name)(defaultData),
 
     // type (debug purpose)
-    { krfType: type },
+    {
+      krfType: type,
+      krfDefaultData: defaultData,
```
Comment (fabienjuif) [inline src/index.js]: please no, defaultData can be huge sometimes

File: src/index.spec.js [hunk 3]
```diff
@@ -135,16 +135,40 @@ describe('index', () => {
...
+      expect(simple.array({ name: 'array' }).krfDefaultData).toMatchSnapshot()
+    })
+  })
+
+  describe('suffix of simpleObject factory', () => {
+    it('should make a simple object init to empty', () => {
+      expect(simpleObject({ name: 'object' }).krfDefaultData).toMatchSnapshot()
+    })
+    it('should make a simple string init to blank', () => {
+      expect(simpleObject({ name: 'object', defaultData: '' }).krfDefaultData).toMatchSnapshot()
```
Comment (fabienjuif) [inline src/index.spec.js]: `name: 'string'` ?

File: src/__snapshots__/index.spec.js.snap (hunk 1 above)
Comment (guillaumecrespel) [reply src/__snapshots__/index.spec.js.snap]: I fix that

File: TYPES.md [hunk 4]
```diff
@@ -100,6 +103,70 @@ Parameters are :
...
+
+export default simple(/* options */)
+```
+
+or
+
+```es6
+import factory from 'k-redux-factory'
+
+export default factory({ type: 'simpleObject', /* other options */ })
```
Comment (fabienjuif) [inline TYPES.md]: type: 'simple' works ?
type: 'simple.bool' works ?

File: TYPES.md [hunk 5]
```diff
@@ -100,6 +103,70 @@ Parameters are :
...
+```
+
+There are multiple factories signatures, take you favorite between:
+ - `simple(middlewares)(options)` : this is a `simple` factory with middlewares
+ - `simple(options)` : this is a `simple` factory without middlewares
+ - `simple.bool(options)` : this is a `simple boolean` factory initialize to `false`
+ - `simple.string(options)` : this is a `simple string` factory initialize to `''`
+ - `simple.array(options)` : this is a `simple array` factory initialize to `[]`
```
Comment (fabienjuif) [inline TYPES.md]: we should add `simple.object`

Comment (guillaumecrespel) [reply TYPES.md]: by default, `simple()` init with empty object.
you think it's important to add this shorter ?

File: TYPES.md (hunk 4 above)
Comment (guillaumecrespel) [reply TYPES.md]: Does not work at the moment.
But your purpose is : 
 - [] rename folder `simpleObject` to `simple`
 - [] rename type `simpleObject` to `simple`
 - [] add type for each shorter ?
 - [] keep on deprecated type `simpleObject`
?

File: TYPES.md (hunk 5 above)
Comment (EmrysMyrddin) [reply TYPES.md]: For me yes, it's better to be explicit about this since the name doesn't contains 'object' anymore.

Comment (fabienjuif) [reply TYPES.md]: Yes, simple can do a simple.object per default, but we have to add it explicitly

File: TYPES.md (hunk 4 above)
Comment (fabienjuif) [reply TYPES.md]: - Yes you can rename folder and imports and file
 - You can add type: 'simple.xxx' to the factory
 - You can keep simpleObject

File: src/types/index.js [hunk 6]
```diff
@@ -1,2 +1,3 @@
 export { default as keyValue } from './keyValue/index'
-export { default as simpleObject } from './simpleObject/index'
+export { default as simple } from './simple/index'
+export { default as simpleObject } from './simple/index'
```
Comment (fabienjuif) [inline src/types/index.js]: You export it but don't use it in the factory file (src/index.js)

Comment (guillaumecrespel) [reply src/types/index.js]: yes, but it's a simple way to keep `simpleObject`.
it's use at this line : https://github.com/alakarteio/k-redux-factory/pull/119/files#diff-1fdf421c05c1140f6d71444ea2b27638R28

Comment (fabienjuif) [reply src/types/index.js]: OK I see 👍

File: TYPES.md (hunk 5 above)
Comment (fabienjuif) [reply TYPES.md]: @guillaumecrespel can you update the readme please ?

**Pull request merged**